prov/verbs/src/verbs_info.c

```diff
@@ -982,6 +987,122 @@ static int vrb_ifa_rdma_info(const struct ifaddrs *ifa, char **dev_name,
 	return ret;
 }
 
+int vrb_get_port_space(const struct fi_info *info)
+{
+	if (info != NULL && info->addr_format == FI_SOCKADDR_IB)
+		return RDMA_PS_IB;
+	else
+		return RDMA_PS_TCP;
+}
+
+static int vrb_gid_is_null(const union ibv_gid *gid)
+{
+	return !(gid->raw[8]  | gid->raw[9]  | gid->raw[10] | gid->raw[11] |
+	         gid->raw[12] | gid->raw[13] | gid->raw[14] | gid->raw[15]);
+}
+
+void vrb_set_sid(enum rdma_port_space ps,
+        uint16_t port, struct sockaddr_ib *sib)
+{
+        sib->sib_sid = htonll(((uint64_t) ps << 16) + ntohs(port));
+
+        if (ps)
+                sib->sib_sid_mask = htonll(RDMA_IB_IP_PS_MASK);
+        if (port)
```

Comment:
I think we can assert that ps is set if port is given.

Reply:
Done.